compiler/rustc_session/src/options.rs

```diff
@@ -1238,6 +1239,19 @@ mod parse {
         };
         true
     }
+
+    pub(crate) fn parse_solver_proof_tree_condition(
+        slot: &mut SolverProofTreeCondition,
+        v: Option<&str>,
+    ) -> bool {
+        match v {
+            None | Some("always") => *slot = SolverProofTreeCondition::Always,
+            Some("never") => *slot = SolverProofTreeCondition::Never,
+            Some("on_error") => *slot = SolverProofTreeCondition::OnError,
```

Comment:
It seems conventional to use kebab-case for flags in the compiler.

```suggestion
            Some("on-error") => *slot = SolverProofTreeCondition::OnError,
```